src/drivers/ll40ls/LidarLitePWM.cpp

```diff
@@ -0,0 +1,293 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+
+/**
+ * @file LidarLitePWM.h
+ * @author Johan Jansen <jnsn.johan@gmail.com>
+ * @author Ban Siesta <bansiesta@gmail.com>
+ *
+ * Driver for the PulsedLight Lidar-Lite range finders connected via PWM.
+ *
+ * This driver accesses the pwm_input published by the pwm_input driver.
+ */
+
+#include "LidarLitePWM.h"
+#include <stdio.h>
+#include <string.h>
+#include <drivers/drv_hrt.h>
+#include <drivers/drv_pwm_input.h>
+
+/* oddly, ERROR is not defined for c++ */
+#ifdef __cplusplus
+static const int ERROR = -1;
+#endif
+
+LidarLitePWM::LidarLitePWM(const char *path) :
+	CDev("LidarLitePWM", path),
+	_work{},
+	_reports(nullptr),
+	_class_instance(-1),
+	_pwmSub(-1),
+	_pwm{},
+	_distance_sensor_topic(-1),
+	_range{},
+	_sample_perf(perf_alloc(PC_ELAPSED, "ll40ls_pwm_read")),
+	_read_errors(perf_alloc(PC_COUNT, "ll40ls_pwm_read_errors")),
+	_buffer_overflows(perf_alloc(PC_COUNT, "ll40ls_pwm_buffer_overflows")),
+	_sensor_zero_resets(perf_alloc(PC_COUNT, "ll40ls_pwm_zero_resets"))
+{
+}
+
+LidarLitePWM::~LidarLitePWM()
+{
+	stop();
+
+	/* free any existing reports */
+	if (_reports != nullptr) {
+		delete _reports;
+	}
+
+	if (_class_instance != -1) {
+		unregister_class_devname(RANGE_FINDER_BASE_DEVICE_PATH, _class_instance);
+	}
+
+	/* free perf counters */
+	perf_free(_sample_perf);
+	perf_free(_buffer_overflows);
+	perf_free(_sensor_zero_resets);
+}
+
+
+int LidarLitePWM::init()
+{
+	/* do regular cdev init */
+	int ret = CDev::init();
+
+	if (ret != OK) {
+		return ERROR;
+	}
+
+	/* allocate basic report buffers */
+	_reports = new ringbuffer::RingBuffer(2, sizeof(struct distance_sensor_s));
+
+	if (_reports == nullptr) {
+		return ERROR;
+	}
+
+	_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
```

Comment:
The classname still makes sense, because that registers the file descriptor class.
You need to use the orb_subscribe_multi() call in all drivers using the same topic group.